server/datastore/mysql/migrations/tables/20240701113709_VPPDBUpdates.go

```diff
@@ -0,0 +1,80 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240701113709, Down_20240701113709)
+}
+
+func Up_20240701113709(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+-- This table is the VPP equivalent of the host_software_installs table.
+-- It tracks the installation of VPP software on particular hosts.
+CREATE TABLE host_vpp_software_installs (
+	id int(10) unsigned NOT NULL AUTO_INCREMENT,
+	host_id INT(10) UNSIGNED NOT NULL,
+	
+	-- This is the adam_id of the VPP software that's being installed
+	adam_id VARCHAR(16) NOT NULL,
+
+	-- This is the UUID of the MDM command issued to install the software
+	command_uuid VARCHAR(127),
+	user_id UNSIGNED,
+
+	-- This indicates whether or not this was a self-service install
+	self_service TINYINT(1) NOT NULL DEFAULT FALSE,
+	
+	-- This is an ID for the event of "associating" the software with a host. 
+	-- This value comes from the "eventId" field in the response here:
+	-- https://developer.apple.com/documentation/devicemanagement/associate_assets
+	associated_event_id VARCHAR(36),
+
+	created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+	updated_at TIMESTAMP NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
+)`)
+	if err != nil {
+		return fmt.Errorf("failed to create table host_vpp_software_installs: %w", err)
+	}
+
+	_, err = tx.Exec(`
+CREATE TABLE vpp_apps_teams (
+	adam_id VARCHAR(16) NOT NULL,
+	team_id INT(10) UNSIGNED NOT NULL
+)`)
+	if err != nil {
+		return fmt.Errorf("failed to create table vpp_apps_teams: %w", err)
+	}
+
+	_, err = tx.Exec(`
+-- This table is the VPP equivalent of the "software_installers" table.
+-- This table is also used as a cache of the response from the "Get Assets"
+-- Apple endpoint as well as the FleetDM website endpoint which will return
+-- the app metadata.
+-- If an asset has an entry here and an entry in vpp_apps_teams, then it has 
```

Comment:
@mna or @roperzh:

I was re-reading this and I realized that this statement won't necessarily be true if we decide to support adding software to "No team". I have seen customers raise the desire to do so before. Do y'all know if that functionality is on the horizon? should we design around that, or just go with this for now?

Reply:
The approach we've used to support "no team" typically has been with a `0` team_id. More specifically, we use two columns - one called `global_or_team_id` that is `NOT NULL` and is `0` for no team, and one that is `team_id` that is NULLable and is null for no team.

The reason for the dual columns approach is that this allows to use `global_or_team_id` as part of a UNIQUE index constraint (e.g. in the case of `vpp_apps_teams`, the unique key would be `global_or_no_team_id, adam_id`) and also a foreign key on teams (using the `team_id` column). We can't do those two features with a single column (either the unique constraint would not work if only a NULLable column was used, or the FK would not work if a non-nullable column was used because team id 0 does not exist).

Reply:
Ah yeah, makes sense! I think we should add that column here as well.